### PR TITLE
Updated docs for #446

### PR DESCRIPTION
**Pull request by xloem**
A separate PR for my work in #434

### CODE REVIEW

File: advanced-usage.md [hunk 1]
```diff
@@ -204,6 +204,102 @@ assert(r.elapsed <= 1); // Less than one second should have elapsed
...
+
+The callback signature looks like this.
+
+```c++
+  bool readCallback(char* buffer, size_t & length);
+```
+
+Provide the callback with the ReadCallback options object.  Only one read callback may be set.
+When called, `length` is the length of `buffer`.  `buffer` should be filled with data, and `length` updated to how much was filled.
+Return `true` on success, or `false` to cancel the transfer.
```
Comment (COM8) [inline advanced-usage.md:223]:
```suggestion
Return `true` on success, or `false` to **cancel** the transfer.
```

File: advanced-usage.md [hunk 2]
```diff
@@ -204,6 +204,102 @@ assert(r.elapsed <= 1); // Less than one second should have elapsed
...
+
+The callback signature looks like this.
+
+```c++
+  bool headerCallback(std::string data);
+```
+
+Provide the callback with the HeaderCallback options object.  Only one header callback may be set.
+When a header callback is set, the Response object's `header` member will not be filled.
+Return `true` on success, or `false` to cancel the transfer.
```
Comment (COM8) [inline advanced-usage.md:238]:
```suggestion
Return `true` on success, or `false` to **cancel** the transfer.
```

File: advanced-usage.md [hunk 3]
```diff
@@ -204,6 +204,102 @@ assert(r.elapsed <= 1); // Less than one second should have elapsed
...
+
+The callback signature looks like this.
+
+```c++
+  bool writeCallback(std::string data);
+```
+
+Provide the callback with the WriteCallback options object.  Only one write callback may be set.
+When a write callback is set, the Response object's `text` member will not be filled.
+Return `true` on success, or `false` to cancel the transfer.
```
Comment (COM8) [inline advanced-usage.md:253]:
```suggestion
Return `true` on success, or `false` to **cancel** the transfer.
```

File: advanced-usage.md [hunk 4]
```diff
@@ -204,6 +204,102 @@ assert(r.elapsed <= 1); // Less than one second should have elapsed
...
+
+While data is being transferred it will be called very frequently, and during slow periods like when nothing is being transferred it can slow down to about one call per second. The callback gets told how much data `libcurl` will transfer and has transferred, in number of bytes.
+
+The callback signature looks like this.
+
+```c++
+  bool progressCallback(size_t downloadTotal, size_t downloadNow, size_t uploadTotal, size_t uploadNow);
+```
+
+The values are in bytes.  Return `true` to continue the transfer, and `false` to cancel it.
```
Comment (COM8) [inline advanced-usage.md:265]:
```suggestion
The values are in bytes.  Return `true` to continue the transfer, and `false` to **cancel** it.
```

File: advanced-usage.md [hunk 5]
```diff
@@ -204,6 +204,102 @@ assert(r.elapsed <= 1); // Less than one second should have elapsed
...
+
+Here is an example of using the callback.
+
+```c++
+int main(int argc, char **argv) {
```
Comment (COM8) [inline advanced-usage.md:270]:
```suggestion
int main(int argc, char** argv) {
```

**Review by COM8 — CHANGES_REQUESTED** [review]
Nice documentation. Thanks!
Only a few things and we are ready to go here.

**Comment by xloem** [comment]
Minor changes look great.  Feel free to add them in yourself, or I will in some time.

**Review by COM8 — APPROVED** [review]

**Pull request merged**